File: src/main/java/com/fatec/destino/dto/compra/CompraRequestDTO.java
package com.fatec.destino.dto.compra;

import com.fatec.destino.util.model.compra.Metodo;
import com.fatec.destino.util.model.compra.Processador;

import java.util.UUID;

public record CompraRequestDTO(
        UUID usuarioId,
        long pacoteId,
        Metodo metodo,          // VISTA, PARCELADO
        Processador processador, // VISA, MASTERCARD, PIX
        int parcelas            // 1 a 12
) {
}

File: src/main/kotlin/com/fatec/destino/dto/compra/CompraRequestDTO.kt
package com.fatec.destino.dto.compra

import com.fatec.destino.util.model.compra.Metodo
import com.fatec.destino.util.model.compra.Processador
import java.util.UUID

data class CompraRequestDTO(
    var usuarioId : UUID,
    var pacoteId: Long,
    var metodo: Metodo,
    var processador : Processador,
    var parcelas : Int
)
